### DISCUSSION TITLE
Debugging Aleph apps in VSCode

### DISCUSSION THREAD

**Discussion by aadamsx** [alephjs/aleph.js#237] [Q&A]
How are you debugging Aleph inside Visual Studio code? In the debugger window/tab?

NOTE: I'm referring to debugging my application that's running inside the Aleph framework, not actually debugging Aleph itself (which I'm also looking at now).

when I try ```Deno: Run``` with default Deno settings I get:

```
Note: Using the "preview" debug extension
/usr/local/bin/deno run --unstable --inspect --allow-all main.ts
Debugger listening on ws://127.0.0.1:9229/ws/374009d0-83a0-43ef-ac8b-6b036bb2dc0a
error: Cannot resolve module "file:///Users/aadams/Projects/mserv-mail/main.ts".
Process exited with code 1
```

These are the configuration files under the \.vscode directory inside the Aleph project

```
\\ .vscode\launch.json
{
  // Use IntelliSense to learn about possible attributes.
  // Hover to view descriptions of existing attributes.
  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Deno: Run",
      "request": "launch",
      "type": "pwa-node",
      "program": "main.ts",
      "cwd": "${workspaceFolder}",
      "runtimeExecutable": "deno",
      "runtimeArgs": [
        "run",
        "--unstable",
        "--inspect",
        "--allow-all"
      ],
      "attachSimplePort": 9229
    }
  ]
}
```

```json
\\ .vscode\settings.json
{
  "deno.enable": true,
  "deno.unstable": true,
  "deno.importMap": "./import_map.json"
}
```

```json
\\ .vscode\extensions.json
{
  "recommendations": [
    "denoland.vscode-deno"
  ]
}
```

---
I then tried to modify the launch.json file like the following without success:

```
// launch.json

  "configurations": [
    {
      "name": "Aleph: Run",
      "request": "launch",
      "type": "pwa-node",
      "runtimeExecutable": "aleph",
      "runtimeArgs": [
        "dev",
        "~/Projects/mserv-mail/",
        "-L",
        "debug"
      ],
      "attachSimplePort": 8081
    },

...

```

```
Error processing launch: Error: Could not connect to debug target at http://127.0.0.1:8081: Promise was canceled
at e (/Users/aadams/.vscode/extensions/ms-vscode.js-debug-nightly-2021.3.3017/src/extension.js:1:115007)
at runMicrotasks (<anonymous>)
at processTicksAndRejections (internal/process/task_queues.js:97:5)
at async t (/Users/aadams/.vscode/extensions/ms-vscode.js-debug-nightly-2021.3.3017/src/extension.js:59:61879)
at async P.launch (/Users/aadams/.vscode/extensions/ms-vscode.js-debug-nightly-2021.3.3017/src/extension.js:1:135988)
at async t.Binder.captureLaunch (/Users/aadams/.vscode/extensions/ms-vscode.js-debug-nightly-2021.3.3017/src/extension.js:59:152076)
at async t.Binder._launch (/Users/aadams/.vscode/extensions/ms-vscode.js-debug-nightly-2021.3.3017/src/extension.js:59:151627)
at async Promise.all (index 3)
at async t.Binder._boot (/Users/aadams/.vscode/extensions/ms-vscode.js-debug-nightly-2021.3.3017/src/extension.js:59:150887)
at async t.default._onMessage (/Users/aadams/.vscode/extensions/ms-vscode.js-debug-nightly-2021.3.3017/src/extension.js:1:88464)

Note: Using the "preview" debug extension
/Users/aadams/.deno/bin/aleph dev ~/Projects/mserv-mail/ -L debug
Process exited with code 1

```

I can debug Deno code fine, it's Aleph that I'm having issues with.  

Any help is appricated.

**Comment by aadamsx** [comment]
@shadowtime2000 @ije do any one of you use VS Code?  This must be just a settings.json change right?  It would be a huge help if someone could just send over their setup please!

**Reply by shadowtime2000** [reply]
I use VS Code, afaik ije does too. Honestly I don't see a need for a debugging ability in a while. One part is personal bias because I have rarely actually found those tools useful, and also I think it is more important to work on stabilizing the current system instead of just improving DX.

**Reply by aadamsx** [reply]
@shadowtime2000 ... so, as a os-`dev`, you're building a web-`dev` tool/framework/lib, and don't think `dev`-experiance is important... whelp!

And what do you use, `console.log` everywhere?  Yikes!

**Reply by shadowtime2000** [reply]
@aadamsx That isn't what I meant. DX is extremely important. I just think the DX is fine where it is and right now we should work on stabilising the bundler and compiler before improving it further. 

Yeah that is basically what I do. Works pretty fine for me 🤷

**Comment by aadamsx** [comment]
Start the server:

`$ aleph dev -reload`

INFO Server ready on http://localhost:8084/

Then try to attach to the process in the `launch.json`:

```
  "configurations": [
    {
      "type": "pwa-node",
      "name": "Aleph: Run",
      "request": "attach",
      "port": 8084
    },
```

But no chance.  Is anyone out there got this working in Aleph yet? 

Seems like few to no answers on this dicussion board these days, can anyone point me to a good spot to answer this question?  Would even stackoverflow even get any answers on Aleph I wonder?

**Comment by aadamsx** [comment]
It's offical, the very first SO question on Aleph.js, here's proof:

![image](https://user-images.githubusercontent.com/5341158/113242585-f1798000-9276-11eb-958f-43b3e6caed47.png)

Let's see if I get any hits: https://stackoverflow.com/questions/66898338/debugging-aleph-apps-in-vscode
